lyon_path: tests failing on `serialization` feature

With `lyon_path`, while running the test suite of the `serialization` feature for the Debian package, I have these errors:

**Command**
```
/usr/share/cargo/bin/cargo-auto-test lyon_path 1.0.5 --all-targets --no-default-features --features serialization
```

which is equivalent for lyon_path to:
```
cargo test --all-targets --no-default-features --features serialization
```

**Errors**
```
error[E0277]: the trait bound `Box<[u32]>: Serialize` is not satisfied
error[E0277]: the trait bound `Box<[u32]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[u32]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[u32]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Serialize` is not satisfied
error[E0277]: the trait bound `Box<[Verb]>: Serialize` is not satisfied
error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[Verb]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[Verb]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Deserialize<'_>` is not satisfied
error[E0277]: the trait bound `Box<[Verb]>: Deserialize<'_>` is not satisfied
```

**Full log**
```
   Compiling lyon_path v1.0.5 (/usr/share/cargo/registry/lyon_path-1.0.5)
     Running `CARGO=/usr/bin/cargo CARGO_CRATE_NAME=lyon_path CARGO_MANIFEST_DIR=/usr/share/cargo/registry/lyon_path-1.0.5 CARGO_PKG_AUTHORS='Nicolas Silva <nical@fastmail.com>' CARGO_PKG_DESCRIPTION='Types and utilities to store, build and iterate over 2D paths.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=lyon_path CARGO_PKG_README=README.md CARGO_PKG_REPOSITORY='https://github.com/nical/lyon' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=1.0.5 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=5 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 CARGO_RUSTC_CURRENT_DIR=/usr/share/cargo/registry/lyon_path-1.0.5 LD_LIBRARY_PATH=/tmp/tmp.fvKE9etDXY/target/debug/deps rustc --crate-name lyon_path --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test --cfg 'feature="serde"' --cfg 'feature="serialization"' --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values("default", "serde", "serialization", "std"))' -C metadata=c3c0c856aa0ccca1 -C extra-filename=-c3c0c856aa0ccca1 --out-dir /tmp/tmp.fvKE9etDXY/target/x86_64-unknown-linux-gnu/debug/deps --target x86_64-unknown-linux-gnu -C incremental=/tmp/tmp.fvKE9etDXY/target/x86_64-unknown-linux-gnu/debug/incremental -L dependency=/tmp/tmp.fvKE9etDXY/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/tmp/tmp.fvKE9etDXY/target/debug/deps --extern lyon_geom=/tmp/tmp.fvKE9etDXY/target/x86_64-unknown-linux-gnu/debug/deps/liblyon_geom-ea3953d9724c317f.rlib --extern num_traits=/tmp/tmp.fvKE9etDXY/target/x86_64-unknown-linux-gnu/debug/deps/libnum_traits-ce620396c53b909c.rlib --extern serde=/tmp/tmp.fvKE9etDXY/target/x86_64-unknown-linux-gnu/debug/deps/libserde-901b90f2066de312.rlib -C debuginfo=2 -C strip=none --cap-lints warn -C linker=x86_64-linux-gnu-gcc -C link-arg=-Wl,-z,relro --remap-path-prefix /usr/share/cargo/registry/lyon_path-1.0.5=/usr/share/cargo/registry/lyon-path-1.0.5 --remap-path-prefix /tmp/tmp.fvKE9etDXY/registry=/usr/share/cargo/registry`
error[E0277]: the trait bound `Box<[u32]>: Serialize` is not satisfied
    --> src/commands.rs:155:46
     |
155  | #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
     |                                              ^^^^^^^^^ the trait `Serialize` is not implemented for `Box<[u32]>`
156  | pub struct PathCommands {
157  |     cmds: Box<[u32]>,
     |     ---- required by a bound introduced by this call
     |
     = note: for local types consider adding `#[derive(serde::Serialize)]` to your `Box<[u32]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Serialize`:
               &'a T
               &'a mut T
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 140 others
note: required by a bound in `serde::ser::SerializeStruct::serialize_field`
    --> /usr/share/cargo/registry/serde-1.0.210/src/ser/mod.rs:1867:21
     |
1865 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1866 |     where
1867 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Box<[u32]>: Deserialize<'_>` is not satisfied
    --> src/commands.rs:157:11
     |
157  |     cmds: Box<[u32]>,
     |           ^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[u32]>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[u32]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 153 others
note: required by a bound in `next_element`
    --> /usr/share/cargo/registry/serde-1.0.210/src/de/mod.rs:1732:12
     |
1730 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1731 |     where
1732 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `Box<[u32]>: Deserialize<'_>` is not satisfied
    --> src/commands.rs:157:11
     |
157  |     cmds: Box<[u32]>,
     |           ^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[u32]>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[u32]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 153 others
note: required by a bound in `next_value`
    --> /usr/share/cargo/registry/serde-1.0.210/src/de/mod.rs:1871:12
     |
1869 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1870 |     where
1871 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

error[E0277]: the trait bound `Box<[u32]>: Deserialize<'_>` is not satisfied
   --> src/commands.rs:155:57
    |
155 | #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
    |                                                         ^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[u32]>`
    |
    = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[u32]>` type
    = note: for types from other crates check whether the crate offers a `serde` feature flag
    = help: the following other types implement trait `Deserialize<'de>`:
              &'a [u8]
              &'a str
              ()
              (T,)
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
            and 153 others
note: required by a bound in `serde::__private::de::missing_field`
   --> /usr/share/cargo/registry/serde-1.0.210/src/private/de.rs:25:8
    |
23  | pub fn missing_field<'de, V, E>(field: &'static str) -> Result<V, E>
    |        ------------- required by a bound in this function
24  | where
25  |     V: Deserialize<'de>,
    |        ^^^^^^^^^^^^^^^^ required by this bound in `missing_field`
    = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Serialize` is not satisfied
    --> src/path.rs:75:46
     |
75   | #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
     |                                              ^^^^^^^^^ the trait `Serialize` is not implemented for `Box<[Point2D<f32, UnknownUnit>]>`
76   | pub struct Path {
77   |     points: Box<[Point]>,
     |     ------ required by a bound introduced by this call
     |
     = note: for local types consider adding `#[derive(serde::Serialize)]` to your `Box<[Point2D<f32, UnknownUnit>]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Serialize`:
               &'a T
               &'a mut T
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 140 others
note: required by a bound in `serde::ser::SerializeStruct::serialize_field`
    --> /usr/share/cargo/registry/serde-1.0.210/src/ser/mod.rs:1867:21
     |
1865 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1866 |     where
1867 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Box<[Verb]>: Serialize` is not satisfied
    --> src/path.rs:75:46
     |
75   | #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
     |                                              ^^^^^^^^^ the trait `Serialize` is not implemented for `Box<[Verb]>`
...
78   |     verbs: Box<[Verb]>,
     |     ----- required by a bound introduced by this call
     |
     = note: for local types consider adding `#[derive(serde::Serialize)]` to your `Box<[Verb]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Serialize`:
               &'a T
               &'a mut T
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 140 others
note: required by a bound in `serde::ser::SerializeStruct::serialize_field`
    --> /usr/share/cargo/registry/serde-1.0.210/src/ser/mod.rs:1867:21
     |
1865 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1866 |     where
1867 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Deserialize<'_>` is not satisfied
    --> src/path.rs:77:13
     |
77   |     points: Box<[Point]>,
     |             ^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[Point2D<f32, UnknownUnit>]>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[Point2D<f32, UnknownUnit>]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 154 others
note: required by a bound in `next_element`
    --> /usr/share/cargo/registry/serde-1.0.210/src/de/mod.rs:1732:12
     |
1730 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1731 |     where
1732 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `Box<[Verb]>: Deserialize<'_>` is not satisfied
    --> src/path.rs:78:12
     |
78   |     verbs: Box<[Verb]>,
     |            ^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[Verb]>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[Verb]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 154 others
note: required by a bound in `next_element`
    --> /usr/share/cargo/registry/serde-1.0.210/src/de/mod.rs:1732:12
     |
1730 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1731 |     where
1732 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Deserialize<'_>` is not satisfied
    --> src/path.rs:77:13
     |
77   |     points: Box<[Point]>,
     |             ^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[Point2D<f32, UnknownUnit>]>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[Point2D<f32, UnknownUnit>]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 154 others
note: required by a bound in `next_value`
    --> /usr/share/cargo/registry/serde-1.0.210/src/de/mod.rs:1871:12
     |
1869 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1870 |     where
1871 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

error[E0277]: the trait bound `Box<[Verb]>: Deserialize<'_>` is not satisfied
    --> src/path.rs:78:12
     |
78   |     verbs: Box<[Verb]>,
     |            ^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[Verb]>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[Verb]>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 154 others
note: required by a bound in `next_value`
    --> /usr/share/cargo/registry/serde-1.0.210/src/de/mod.rs:1871:12
     |
1869 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1870 |     where
1871 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

error[E0277]: the trait bound `Box<[Point2D<f32, UnknownUnit>]>: Deserialize<'_>` is not satisfied
  --> src/path.rs:75:57
   |
75 | #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
   |                                                         ^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[Point2D<f32, UnknownUnit>]>`
   |
   = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[Point2D<f32, UnknownUnit>]>` type
   = note: for types from other crates check whether the crate offers a `serde` feature flag
   = help: the following other types implement trait `Deserialize<'de>`:
             &'a [u8]
             &'a str
             ()
             (T,)
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
           and 154 others
note: required by a bound in `serde::__private::de::missing_field`
  --> /usr/share/cargo/registry/serde-1.0.210/src/private/de.rs:25:8
   |
23 | pub fn missing_field<'de, V, E>(field: &'static str) -> Result<V, E>
   |        ------------- required by a bound in this function
24 | where
25 |     V: Deserialize<'de>,
   |        ^^^^^^^^^^^^^^^^ required by this bound in `missing_field`
   = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Box<[Verb]>: Deserialize<'_>` is not satisfied
  --> src/path.rs:75:57
   |
75 | #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
   |                                                         ^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Box<[Verb]>`
   |
   = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Box<[Verb]>` type
   = note: for types from other crates check whether the crate offers a `serde` feature flag
   = help: the following other types implement trait `Deserialize<'de>`:
             &'a [u8]
             &'a str
             ()
             (T,)
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
           and 154 others
note: required by a bound in `serde::__private::de::missing_field`
  --> /usr/share/cargo/registry/serde-1.0.210/src/private/de.rs:25:8
   |
23 | pub fn missing_field<'de, V, E>(field: &'static str) -> Result<V, E>
   |        ------------- required by a bound in this function
24 | where
25 |     V: Deserialize<'de>,
   |        ^^^^^^^^^^^^^^^^ required by this bound in `missing_field`
   = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: unused import: `num_traits::Float`
  --> src/builder.rs:96:5
   |
96 | use num_traits::Float;
   |     ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0277`.
warning: `lyon_path` (lib test) generated 1 warning
error: could not compile `lyon_path` (lib test) due to 12 previous errors; 1 warning emitted

```



A possible fix would be to use "serde/std" in the feature serialization, instead of "serde" only (in lyon_* package)

--- a/crates/path/Cargo.toml
+++ b/crates/path/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lyon_path"
-version = "1.0.5"
+version = "1.0.6"
 description = "Types and utilities to store, build and iterate over 2D paths."
 authors = ["Nicolas Silva <nical@fastmail.com>"]
 repository = "https://github.com/nical/lyon"

--- a/crates/path/Cargo.toml
+++ b/crates/path/Cargo.toml
@@ -15,7 +15,7 @@ name = "lyon_path"
 [features]
 default = ["std"]
 std = ["lyon_geom/std", "num-traits/std"]
-serialization = ["serde", "lyon_geom/serialization"]
+serialization = ["serde/std", "lyon_geom/serialization"]
 
 [dependencies]
 lyon_geom = { version = "1.0.4", path = "../geom", default-features = false }

